README.md

```diff
@@ -12,23 +12,111 @@ You can find a further details and a more thorough description and discussion of
+1. Create an AWS account. For this post, you configure the required AWS resources in the `us-east-1`, `us-west-2` or `eu-central-1` Region. If you haven’t signed up, complete the following tasks:
 
-    a. Create an account. For instructions, see [Sign Up for AWS](https://lakeformation.aworkshop.io/30-prerequisite/301-sign-up-aws.html).
+    a. Create an account. For instructions, see [Sign Up for AWS](https://aws.amazon.com/resources/create-account/).
     
     b. Create an [AWS Identity and Access Management](https://aws.amazon.com/iam/) (IAM) user. For instructions, see [Create an IAM User](https://lakeformation.aworkshop.io/30-prerequisite/302-create-iam-account.html).
     
-2. Have a Bearer Token associated with your Twitter app. To create a developer account, see [Get started with the Twitter developer platform](https://developer.twitter.com/en/docs/getting-started).
-3. Install [Docker](https://docs.docker.com/get-docker/) on your local machine.
-4. Java 11
+2. Install [Docker](https://docs.docker.com/get-docker/) on your local machine.
+3. Install Java 21 on your local machine.
 
+## Setup
 
+1. Clone this repository:
+   ```
+    git clone https://github.com/aws-samples/amazon-msk-with-apache-kafka-streams-api
+   ```
+2. Deploy the cloud formation template `./misc/cfn-kafka-streams-msk.yml` in your account
+3. Build the applications by running the following command in the root of the project:
+   ```
+   ./gradlew clean build
+   ```
+
+4. Create your Docker images (kafka-streams-msk and twitter-stream-producer):
+   ```
+   docker-compose build
+   ```
+
+5. Retrieve an authentication token and authenticate your Docker client to your registry. Use the following AWS Command Line Interface (AWS CLI) code:
+   ```
+   aws ecr get-login-password --region <<region>> | docker login --username AWS --password-stdin <<account_id>>.dkr.ecr.<<region>>.amazonaws.com
+   ```
+
+6. Tag and push your images to the Amazon ECR repository:
+   ```
+   docker tag kafka-streams-msk:latest  <<account_id>>.dkr.ecr.<<region>>.amazonaws.com/kafka-streams-msk:latest 
+   docker tag twitter-stream-producer:latest  <<account_id>>.dkr.ecr.<<region>>.amazonaws.com/twitter-stream-producer:latest
+   ```
+
+7. Run the following command to push images to your Amazon ECR repositories:
+   ```
+   docker push <<account_id>>.dkr.ecr.<<region>>.amazonaws.com/kafka-streams-msk:latest 
+   docker push <<account_id>>.dkr.ecr.<<region>>.amazonaws.com/twitter-stream-producer:latest
+   ```
+
+8. Ensure that the cloud formation template `cfn-kafka-streams-msk.yml` is deployed
+9. Deploy the cloud formation template `./misc/cfn-ecs-fargate.yml` in your account
+
+
+
+
+## Project Structure
+
+- `/kafka-streams-msk/*`: Kafka Streams implementation including Dockerfile
+- `/misc/`: Contains the cloud formation definition for the ECS and MSK cluster
+- `/twitter-stream-producer/`: Twitter Stream producer
```

Comment:
Can we rename twitter to X everywhere?

Reply:
The name of twitter api still exists, therefore I suggest to leave it for the moment.